### PR TITLE
[INFINITY-2134][INFINITY-2146] Specify virtual network at a service level

### PR DESCRIPTION
**Pull request by elezar**
This PR:
* moves the specification the virtual networks settings (enabled, name, plugins) to the service level.
* renames the settings:
  * `virtual_network_enabled`
  * `virtual_network_name`
  * `virtual_network_plugin_labels`

### CODE REVIEW

File: frameworks/cassandra/universe/config.json [hunk 1]
```diff
@@ -51,11 +51,21 @@
...
           "type": "boolean",
           "default": false
         },
+        "virtual_network_name": {
+          "description": "Name of virtual network to join",
+          "type": "string",
+          "default": "dcos"
+        },
+        "virtual_network_plugin_labels": {
+          "description": "Labels to pass to the virtual network plugin. Comma-separated key:value pairs. For example [k_0:v_0,k_1:v_1,...,k_n:v_n]",
```
Comment (ArtRand) [inline frameworks/cassandra/universe/config.json]: request https://jira.mesosphere.com/browse/INFINITY-2146 was to remove the brackets around the example e.g.: `[k_0:v_0,k_1:v_1,...,k_n:v_n]` -> `k_0:v_0,k_1:v_1,...,k_n:v_n`

Comment (elezar) [inline frameworks/cassandra/universe/config.json]: Ah, so if I do it here, then we can close that too?

**Comment by elezar** [comment]
A quick question from my side: Is there a way to "inherit" service specs in some way? For the moment, it seems like changes like these require that one make the edits to **each** of the framework-specific files. This of course is a lot of manual work, and also means that there is a much higher chance of typos/bugs creeping in.

File: frameworks/cassandra/universe/config.json (hunk 1 above)
Comment (elezar) [inline frameworks/cassandra/universe/config.json]: OK, another question @ArtRand. I see from `AttributeStringUtilsTest` that for other key-value like parameters we sometimes use `;` as a separator instead of `,`. Should I update the networking spec to match that too?

**Comment by ArtRand** [comment]
@elezar you'd have to do it with some kind of templating I think, @nickbp would be the best person to ask. Thanks for the changes. LGTM!

**Comment by benclarkwood** [comment]
@elezar I have thoughts around this that hinge on having a single uber yaml file that spits out universe packages.

**Comment by benclarkwood** [comment]
For platform wide features like CNI, the setup can be injected in a system like that.

**Comment by benclarkwood** [comment]
A bridge to that _could_ be a template for the "core" of a service spec. But I'm hesitant to add in yet more templating without a common control mechanism.

**Comment by nickbp** [comment]
I think inheritance would make sense eventually, though I think we'd get bigger payoff having some sort of build-time templating of config.json/marathon.json.mustache before worrying too much about svc.yml. If we want to clean up duplicate code, config.json is a good plate to start.

File: frameworks/cassandra/universe/config.json [hunk 2]
```diff
@@ -51,11 +51,21 @@
...
           "default": "serial"
         },
-        "virtual_networks": {
+        "virtual_network_enabled": {
           "description": "Enable DC/OS Virtual Networking",
           "type": "boolean",
           "default": false
         },
+        "virtual_network_name": {
+          "description": "Name of virtual network to join",
```
Comment (benclarkwood) [inline frameworks/cassandra/universe/config.json]: Name of the virtual network to join

(I think? @sascala)

File: frameworks/cassandra/universe/config.json [hunk 3]
```diff
@@ -51,11 +51,21 @@
           "enum": ["serial", "parallel"],
           "default": "serial"
         },
-        "virtual_networks": {
+        "virtual_network_enabled": {
           "description": "Enable DC/OS Virtual Networking",
```
Comment (benclarkwood) [inline frameworks/cassandra/universe/config.json]: Should be "Enable virtual networking"

**Comment by elezar** [comment]
I've updated the descriptions as requested @benclarkwood. As soon as the CI is green, we can merge these (for the videos)

**Comment by elezar** [comment]
**TL;DR:** The virtual network settings are applied correctly to the template service.

I have tested these changes with the following steps:
1. Created a json file `network-config.json`:
```
{
    "service": {
        "virtual_network_enabled": true,
        "virtual_network_name": "dcos",
        "virtual_network_plugin_labels": "key1:value1,key2:value2"
    }
}
```
2. Built an aws package from my local changes:
```
cd frameworks/template && ./build.sh aws
```
3. Install the package (using the stub universe):
```
dcos package install --yes template --options=network-config.json
```
4. Check the config:
```
$ dcos template describe
{
  "node": {
    "count": 1,
    "cpus": 0.1,
    "disk": 25,
    "disk_type": "ROOT",
    "mem": 252,
    "placement_constraint": ""
  },
  "service": {
    "log_level": "INFO",
    "name": "template",
    "principal": "",
    "secret_name": "",
    "sleep": 1000,
    "user": "root",
    "virtual_network_enabled": true,
    "virtual_network_name": "dcos",
    "virtual_network_plugin_labels": "key1:value1,key2:value2"
  }
}
```
5. Check the state of the running task:
```
$ dcos task template-0-node --json

[
  {
    "container": {
      "type": "MESOS"
    },
    "executor_id": "template__a26c1542-40e0-4f25-bc84-589131bb1b23",
    "framework_id": "3167b565-6148-4715-860e-5e4c981551d6-0003",
    "id": "template-0-node__6e2f458e-2da5-46f0-bb89-88b84f986bde",
    "labels": [
      {
        "key": "goal_state",
        "value": "RUNNING"
      },
      {
        "key": "index",
        "value": "0"
      },
      {
        "key": "offer_attributes",
        "value": ""
      },
      {
        "key": "offer_hostname",
        "value": "10.0.0.145"
      },
      {
        "key": "target_configuration",
        "value": "cd32717d-02fd-42e3-a777-ad565c11d684"
      },
      {
        "key": "task_type",
        "value": "template"
      }
    ],
    "name": "template-0-node",
    "resources": {
      "cpus": 0.1,
      "disk": 25.0,
      "gpus": 0.0,
      "mem": 252.0
    },
    "role": "template-role",
    "slave_id": "3167b565-6148-4715-860e-5e4c981551d6-S3",
    "state": "TASK_RUNNING",
    "statuses": [
      {
        "container_status": {
          "container_id": {
            "parent": {
              "value": "d669cdeb-3218-4612-97f6-d50facb9f8d4"
            },
            "value": "85064bf5-68cb-45e2-84c6-50f6169ed153"
          },
          "network_infos": [
            {
              "ip_addresses": [
                {
                  "ip_address": "9.0.4.2",
                  "protocol": "IPv4"
                }
              ],
              "labels": [
                {
                  "key": "key1",
                  "value": "value1"
                },
                {
                  "key": "key2",
                  "value": "value2"
                }
              ],
              "name": "dcos"
            }
          ]
        },
        "state": "TASK_RUNNING",
        "timestamp": 1502703190.39002
      }
    ],
    "user": "root"
  }
]
```

Checking in the cluster UI, the task has also joined the network.

**Comment by elezar** [comment]
The last two commits contain only documentation changes.

**Comment by elezar** [comment]
Any comments on the doc changes?

**Comment by benclarkwood** [comment]
I'm good with the docs changes. @sascala to provide final say on those.

File: docs/pages/operations-guide.md [hunk 4]
```diff
@@ -297,23 +297,23 @@ pods:
       ...
 ```
 
-When a pod is on the `dcos` overlay network:
+When a pod is on a virtual network such as the `dcos`:
   * Every pod gets its own IP address and its own array of ports.
   * Pods do not use the ports on the host machine.
   * Pod IP addresses can be resolved with the DNS: `<task_name>.<service_name>.autoip.dcos.thisdcos.directory`.
 
-Specifying that pods join the `dcos` overlay network has the following indirect effects:
```
Comment (sascala) [inline docs/pages/operations-guide.md]: typo
